dali/python/nvidia/dali/external_source.py

```diff
@@ -159,11 +185,16 @@ class ExternalSource():
 
     The source is polled for data (via a call ``source()`` or ``next(source)``)
     when the pipeline needs input for the next iteration. Depending on the value of ``num_outputs``,
-    the source can supply one or more data batches. If ``num_outputs`` is not set, the ``source``
-    is expected to return one batch. If this value is specified, the data is expected to a be tuple,
-    or list, where each element corresponds to respective return value of the external_source.
-    If the source is a callable that accepts a positional argument, it is assumed to be the current
+    the source can supply one or more data items. The data item can be a whole batch (default) or
+    a single batch entry (when ``batch==False``). If ``num_outputs`` is not set, the ``source``
+    is expected to return one item (a batch or a sample). If this value is specified (even if its
+    value is 1), the data is expected to a be tuple, or list, where each element corresponds to
+    respective return value of the external_source.
+
+    A per-batch source may accept one positional argument. If it does, it is the current iteration
     iteration number and consecutive calls will be ``source(0)``, ``source(1)``, and so on.
+    A per-sample source may accept a :class:`nvidia.dali.types.SampleInfo` structure.
```

Comment:
May or should?

Reply:
May = if it takes a parameter, then it's that structure. But it may also take no parameters at all.